tests/test_urlhaus.py

```diff
@@ -0,0 +1,469 @@
+        (
+            "url",
+            {
+                "threat": "malware_download",
+                "url_status": "online",
+                "tags": ["phishing", "fake_update"],
+                "payloads": [
+                    {
+                        "file_type": "PE32 executable (GUI) Intel 80386, for MS Windows",
+                        "filename": "update.exe",
+                        "response_md5": "a1b2c3d4",
+                        "signature": "Trojan.GenericKD.33632270",
+                        "virustotal": {
+                            "percent": 80,
+                        },
+                    }
+                ],
+            },
+            {
+                "source": "URLhaus",
+                "url": "https://malicious.com",
+                "description": (
+                    "Connecting to a malicious URL https://malicious.com. "
+                    "Detected by: URLhaus threat: malware_download, URL status: online, "
+                    "tags: phishing fake_update, the file hosted in this url is of type: "
+                    "PE32 executable (GUI) Intel 80386, for MS Windows, filename: update.exe "
+                    "md5: a1b2c3d4 signature: Trojan.GenericKD.33632270. and was marked by "
+                    "80% of virustotal's AVs as malicious"
+                ),
+                "threat_level": 80,
+                "tags": "phishing fake_update",
+            },
+            "https://malicious.com",
+        ),
+        # Testcase2: Testing MD5 hash parsing all fields present
+        (
+            "md5_hash",
+            {
+                "file_type": "PE32 executable (console) Intel 80386, for MS Windows",
+                "filename": "malware.exe",
+                "signature": "Trojan.Win32.Generic!BT",
+                "virustotal": {
+                    "percent": 95,
+                },
+            },
+            {
+                "blacklist": "URLhaus",
+                "threat_level": 95,
+                "tags": "Trojan.Win32.Generic!BT",
+                "file_type": "PE32 executable (console) Intel 80386, for MS Windows",
+                "file_name": "malware.exe",
+            },
+            "a1b2c3d4",
+        ),
+        # Testcase 3: MD5 hash parsing  missing 'virustotal'
+        (
+            "md5_hash",
+            {
+                "file_type": "PE32 executable (console) Intel 80386, for MS Windows",
+                "filename": "malware.exe",
+                "signature": "Trojan.Win32.Generic!BT",
+            },
+            {
+                "blacklist": "URLhaus",
+                "threat_level": False,
+                "tags": "Trojan.Win32.Generic!BT",
+                "file_type": "PE32 executable (console) Intel 80386, for MS Windows",
+                "file_name": "malware.exe",
+            },
+            "a1b2c3d4",
+        ),
+    ],
+)
+@patch.object(URLhaus, "make_urlhaus_request")
+def test_parse_urlhaus_responses(
```

Comment:
anything that comes after "test_" in the function name should be a function that exists in the class you're testing , right now there is no function called parse_urlhaus_responses()